app.py

```diff
@@ -295,7 +295,18 @@ def valid_request(request: Dict[str, str]) -> str:
     return(auth_status)
 
 
-def get_container(userid: str, request: flask.Request, narrative: str) -> flask.Response:
+def clean_userid( userid: str) -> str:
+    """
+    Takes a normal KBase userid and converts it into a userid that is okay to embed in a rancher servicename
+    """
+    cleaned = userid.replace('_','-')
```

Comment:
print(clean_userid('boris____boris')) doesn't work properly  == boris--boris

should probably do a global replace

not sure why you are messing with "-" . If the username contains a "-" or a "." it should crash, because its not a valid username. Usernames should only be "_"

Reply:
Yeah, that code has been replaced with a regex substitution.